### ISSUE TITLE
.NET Project fails to start when using windows10 TFM, UseCommonOutputDirectory=true, and GenerateRuntimeConfigDevFile=true

### ISSUE THREAD

**Issue by JHJonasHuber** [dotnet/sdk#36474]
### Description

Running dotnet applications will fail with
`The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.`
For specific configurations.

### Reproduction Steps

1. Create a new ConsoleApplication (.NET) and select .NET6 as target framework.
2. Set the Output type to WinExe
3. Set the TargetOS to Windows
4. Set the TargetOS version Version to 10.0.22000.0
5. Set UseCommonOutputDirectory to true
6. Set GenerateRuntimeConfigDevFile to true
7. Set OutputPath to any path.
8. Delete all "obj" and build(debug/release) folders under your Project folder <- **This is important**
9. Clean & rebuild the Project
10. Try to start the ConsoleApplication (With the debugger, so you get the error message)
(Example .csproj file)
```
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0-windows10.0.22000.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <UseCommonOutputDirectory>true</UseCommonOutputDirectory>
    <GenerateRuntimeConfigDevFile>true</GenerateRuntimeConfigDevFile>
    <OutputPath>$(SolutionDir)Build\$(Configuration)\</OutputPath>
  </PropertyGroup>
```

### Expected behavior

Application starts 

### Actual behavior

Process terminates with error message: `The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.`

**Comment by JHJonasHuber** [comment]
Thanks for the explanation of this problem.

So maybe i'm on the completely wrong track to solve my core problem.
I have a large solution (about 115 projects, most c#/.NET, some C++/CLR and a few C++). They all have the same output directory (hey are resulting in ~3 independent executeables (well they share some 3rd party dlls).
So with .NET Framework there were no problems to build them, and create an installer of the produced output.
After switching to .NET6, the build time increased drastically (from 5 minutes to 20). There were a lot of duplicate copies and access violations during copy. So I thought it would be the right way to use the "UseCommonOutputDirectory" setting.
After setting that the build took about 2 minutes. But the output was not useable -> Dependecies to files in NuGet Packages were not copied. So I set "GenerateRuntimeConfigDevFile" -> the dependecies should be resolved (but not copied) when starting the application. I figured out to use "publish" instead of build (C++ projects, with output type "DLL" not supported, but that is another story).  But for publish I can only select one project, but as said my solution produces several executeables.

So in short I want the same build speed as with .NET Framework and all dependecies of the projects, so I can create an installer.
Is "UseCommonOutputDirectoy" the right approach or did I miss something?
